src/AppInstallerCLICore/Workflows/PortableFlow.cpp

```diff
@@ -280,16 +332,116 @@ namespace AppInstaller::CLI::Workflow
+            uninstallEntry.SetValue(PortableValueName::SHA256, Utility::SHA256::ConvertToWideString(context.Get<Execution::Data::HashPair>().second));
+            uninstallEntry.SetValue(PortableValueName::InstallDirectoryCreated, isDirectoryCreated);
+        }
+
+        void RemovePortableExe(Execution::Context& context)
+        {
+            PortableARPEntry uninstallEntry = GetPortableARPEntryForUninstall(context);
+            const auto& targetPath = uninstallEntry[PortableValueName::PortableTargetFullPath];
+
+            if (targetPath.has_value())
+            {
+                const std::filesystem::path& targetPathValue = targetPath.value().GetValue<Value::Type::UTF16String>();
+                const auto& expectedHash = uninstallEntry[PortableValueName::SHA256];
+                std::string expectedHashValue = expectedHash.has_value() ? uninstallEntry[PortableValueName::SHA256].value().GetValue<Value::Type::String>() : "";
+
+                if (std::filesystem::exists(targetPathValue))
+                {
+                    std::ifstream inStream{ targetPathValue, std::ifstream::binary };
+                    const Utility::SHA256::HashBuffer& targetFileHash = SHA256::ComputeHash(inStream);
+                    inStream.close();
+
+                    if (SHA256::AreEqual(SHA256::ConvertToBytes(expectedHashValue), targetFileHash))
+                    {
+                        std::filesystem::remove(targetPathValue);
+                        AICLI_LOG(CLI, Info, << "Successfully deleted portable exe:" << targetPathValue);
+                    }
+                    else
+                    {
+                        if (context.Args.Contains(Execution::Args::Type::HashOverride))
+                        {
+                            context.Reporter.Warn() << Resource::String::PortableHashMismatchOverridden << std::endl;
+                            std::filesystem::remove(targetPathValue);
+                            AICLI_LOG(CLI, Info, << "Successfully deleted portable exe:" << targetPathValue);
+                        }
+                        else
+                        {
+                            context.Reporter.Warn() << Resource::String::PortableHashMismatchOverrideRequired << std::endl;
+                            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_PORTABLE_UNINSTALL_FAILED);
+                        }
+                    }
+                }
+                else
+                {
+                    AICLI_LOG(CLI, Info, << "Portable exe not found; Unable to delete portable exe: " << targetPathValue);
+                }
+            }
+            else
+            {
+                AICLI_LOG(CLI, Info, << "The registry value for [TargetFullPath] does not exist");
+            }
+        }
+
+        void RemoveInstallDirectory(Execution::Context& context)
+        {
+            PortableARPEntry uninstallEntry = GetPortableARPEntryForUninstall(context);
```

Comment:
Why not insert the `PortableARPEntry` into the context and get it out each time rather than reconstructing it repeatedly?

Reply:
Added a PortableARPEntry data to context to be used between tasks